include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -131,6 +187,13 @@ class pool_memory_resource final
    */
   [[nodiscard]] bool supports_get_mem_info() const noexcept override { return false; }
 
+  /**
+   * @brief Get the upstream memory_resource object.
+   *
+   * @return const reference to the upstream memory resource.
+   */
+  [[nodiscard]] const Upstream& upstream_resource() const noexcept { return *upstream_mr_; }
```

Comment:
question: Related to my question above about changing `device_buffer::memory_resource` to return `async_resource_ref` instead of `device_memory_resource*`, why did we add a new function to return the upstream as a `const&` in addition to the `get_upstream` that returns a pointer? In other words, why could we change the return type for `device_buffer::memory_resource` but not `pool_memory_resource::get_upstream`?

Reply:
We need this to forward properties